internal/cache/controller_obj_cache.go

```diff
@@ -1246,13 +1250,17 @@ func (c *AviObjCache) AviPopulateOneVsVipCache(client *clients.AviClient,
 		}
 
 		var vips []string
-		var networkName string
+		var networkNames []string
 		for _, vip := range vsvip.Vip {
 			vips = append(vips, *vip.IPAddress.Addr)
 			if ipamNetworkSubnet := vip.IPAMNetworkSubnet; ipamNetworkSubnet != nil {
 				if networkRef := *ipamNetworkSubnet.NetworkRef; networkRef != "" {
-					if networRefName := strings.Split(networkRef, "#"); len(networRefName) == 2 {
-						networkName = networRefName[1]
+					if networkRefName := strings.Split(networkRef, "#"); len(networkRefName) == 2 {
+						networkNames = append(networkNames, networkRefName[1])
+					}
+					if lib.GetCloudType() == lib.CLOUD_AWS {
```

Comment:
Is there a chance that in AWS we may add two entries for the same network ? Both of the conditions would satisfy.

Reply:
multiple entries of the same network are allowed and will end up creating multiple VIPs in the same network.